[Pei Yee] iP

PeiYee88 opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~Fast~~ ***SUPER*** **FAST** to use

All you need to do is,

1. download it.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

PeiYee88 committed:
changes

PeiYee88 committed:
Level-1

PeiYee88 committed:
Level-2 changes

PeiYee88 committed:
Level 3 - mark as Done

PeiYee88 committed:
Level 4 - added different task classes

PeiYee88 committed:
TextUI Testing

PeiYee88 committed:
Level-5

PeiYee88 committed:
added delete feature

PeiYee88 committed:
test

PeiYee88 committed:
trying textuitest

PeiYee88 committed:
feed

PeiYee88 committed:
Create input.txt

PeiYee88 committed:
Delete input.txt~

PeiYee88 committed:
Delete feed.txt

PeiYee88 committed:
Update runtest.bat

PeiYee88 committed:
Update runtest.sh

PeiYee88 committed:
Delete ACT.TXT

PeiYee88 committed:
Update runtest.bat

PeiYee88 committed:
Update input.txt

PeiYee88 committed:
add input.txt

PeiYee88 committed:
Add save feature

PeiYee88 committed:
store deadlines as time

PeiYee88 committed:
merge

PeiYee88 committed:
refine date feature

PeiYee88 committed:
extract more classes

PeiYee88 committed:
add duke package

PeiYee88 committed:
gradle

PeiYee88 committed:
gradle

PeiYee88 committed:
Merge branch 'add-gradle-support'

PeiYee88 committed:
add j-unit

PeiYee88 committed:
build jar

PeiYee88 committed:
add find

PeiYee88 committed:
comply coding standard

PeiYee88 committed:
code to standard

PeiYee88 committed:
comply to coding standard

PeiYee88 committed:
java doc

PeiYee88 committed:
adds documentation

PeiYee88 committed:
add changes to java doc

PeiYee88 committed:
Create PR.md

PeiYee88 commented:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~Fast~~ ***SUPER*** **FAST** to use

All you need to do is,

1. download it.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

PeiYee88 committed:
check style

PeiYee88 committed:
Merge branch 'master' of https://github.com/PeiYee88/ip

vvidday commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,169 @@
+     *
+     * @param file the file to be load.
+     */
+    public void load_task(File file) {
```
Should the method name be in camelCase? (`loadTask`)

vvidday commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,169 @@
+        load_initial(file);
+    }
+
+    private void load_initial(File file) {
```
Same as above - should it be `loadInitial` instead?

vvidday commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,169 @@
+     * @param s string representation of the task.
+     * @return the task that given the description of the string.
+     */
+    public Task stringToTask(String s) {
```
Should the method name be a verb? Maybe something like `getTaskFromString`

vvidday commented on src/main/java/duke/EmptyCommandException.java:
```diff
@@ -0,0 +1,17 @@
+/**
+ * Represents the empty command exception when nothing is feed into the command line.
+ */
+public class EmptyCommandException extends Exception {
```
I like how you have different exceptions and custom error messages for each - really improves the user experience!

vvidday commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,137 @@
+            String command = descriptions[0];
+            File file = new File(filepath);
+            if (command.equals("mark") || command.equals("unmark")) {
+                if (descriptions.length == 1) {
```
I like how you thought about & handled the edge case here! (and the other commands as well!)

vvidday reviewed (commented):
Great job overall, your code is very clean & well documented! Just some very minor issues relating to coding standards, but otherwise the code is of great quality, keep up the great work!

arnav-ag commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,76 @@
+    /**
+     * Minus the count of the tasks.
+     */
+    public void minusCount() {
```
Perhaps this could use a verb as the first part (i.e. subtractCount) - this is a minor issue

arnav-ag commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,112 @@
+     *
+     * @param status the status of the task. [X] for done, [ ] for undone.
+     */
+    public void setStatus(String status) {
```
Would it be better to use a boolean for the status and then you parse that boolean value into "[ ]" or "[X]"? Just in case someone ends up calling this method with some other random string

arnav-ag commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,137 @@
+            } else {
+                throw new InvalidCommandException(command);
+            }
+        } catch (EmptyCommandException e) {
```
Can these statements be compacted to one catch statement? i.e.

```java
catch (Exception e) {
    System.out.println(e.getMessage());
    respond();
}
```

arnav-ag reviewed (commented):
LGTM! Easy to read code with good documentation with very minor corrections suggested 👍

PeiYee88 committed:
add Gui

PeiYee88 committed:
add javadoc

PeiYee88 committed:
Add priority feature

PeiYee88 committed:
Add Assertions

PeiYee88 committed:
Add Code Quality

PeiYee88 committed:
Merge pull request #1 from PeiYee88/branch-A-Assertions

Add Assertions

PeiYee88 committed:
Merge pull request #2 from PeiYee88/branch-A-CodeQuality

Add Code Quality

PeiYee88 committed:
Remove ellipsis

Remove ellipsis when the output is shown on screen.

Ellipsis is printed instead of full output.

Let's modify the label in the dialog box.

Ellipsis being removed can let users see the full description of the output.

PeiYee88 committed:
Add Ui.png

Add Ui.png to showcase my product.

The users can now view how my product design will be like.

The design uses the SpongeBob theme.

PeiYee88 committed:
Update README.md

Add user guide.

PeiYee88 committed:
Add more testing

Add more testing to the following command:
delete, find priority, mark, unmark

Allow users to detect what they have input wrongly.

PeiYee88 committed:
Finalize code for submission

PeiYee88 committed:
Remove comments

PeiYee88 committed:
Resolve method too long issue

PeiYee88 committed:
Change import statements

PeiYee88 committed:
Resolve deprecated gradle issue

Use mainClass instead of mainClassName in application.

PeiYee88 committed:
Update README.md

PeiYee88 committed:
Update and rename README.md to doc/README.md
